kfs-core/src/main/java/edu/arizona/kfs/pdp/service/impl/PayeeAchAccountServiceImpl.java

```diff
@@ -0,0 +1,266 @@
+		if (StringUtils.isBlank(groupId)) {
+			return null;
+		}
+		
+		List<String> overrideIds = getGroupService().getDirectMemberPrincipalIds(groupId);
+		return overrideIds;
+	}
+	
+	protected List<String> addEmplIdsToPrincipalIds(List<String> principalIds) {
+		if (principalIds == null) {
+			return null;
+		}
+		
+		List<String> emplIds = new ArrayList<String>();
+		
+		Person person;
+		String emplId;
+		for (String principalId : principalIds) {
+			person = personService.getPerson(principalId);
+			emplId = person.getEmployeeId();
+			if (StringUtils.isNotBlank(emplId)) {
+				if (!emplIds.contains(emplId)) {
+					emplIds.add(emplId);
+				}
+			}
+		}
+		
+		emplIds.addAll(principalIds);
+		return emplIds;
+	}
+	
+	protected Integer resetRecordsOfType(String inputFileName) {
+		Integer recordsReset = new Integer(0);
+		
+		String payeeType = validateTypeUsedInFile(inputFileName);
+		if (StringUtils.isBlank(payeeType)) {
+			throw new RuntimeException("PayeeType was blank from file [" + inputFileName + "]. This should never happen.");
+		}
+		
+		List<String> overridePrincipalIds = getPrincipalIdsInOverrideGroup();
+		
+		List<String> overridePrincipalAndEmplIds = addEmplIdsToPrincipalIds(overridePrincipalIds);
+		
+		recordsReset = removePayeeAchAccountsByPayeeType(payeeType, overridePrincipalAndEmplIds);
+		
+		return recordsReset;
+	}
+	
+	// this method will remove those accounts that match the PayeeIdentifierType on the File provided but not the list of payeeIdNumber's found on the BANKING_INFORMATION_OVERRIDE_GROUP group
+	private Integer removePayeeAchAccountsByPayeeType(String payeeType, List<String> overridePrincipalAndEmplIds) {
+		List<PayeeACHAccount> payeesToDelete = new ArrayList<PayeeACHAccount>();
+		
+		Map<String, String> criteria = new HashMap<String, String>();
+		criteria.put(PdpPropertyConstants.PAYEE_IDENTIFIER_TYPE_CODE, payeeType);
+		Collection<PayeeACHAccount> fetchedPayees = businessObjectService.findMatching(PayeeACHAccount.class, criteria);
+		
+		if (overridePrincipalAndEmplIds == null || overridePrincipalAndEmplIds.isEmpty()) {
+			payeesToDelete.addAll(fetchedPayees);
+		} else {
+			for (PayeeACHAccount payee : fetchedPayees) {
+				if (!overridePrincipalAndEmplIds.contains(payee.getPayeeIdNumber())) {
+					payeesToDelete.add(payee);
+				}
+			}
+		}
+		
+		deletePayees(payeesToDelete);
+
+		return payeesToDelete.size();
+	}
+	
+	private void deletePayees(List<PayeeACHAccount> payees) {
+		if (payees == null || payees.isEmpty()) {
+			return;
+		}
+		
+		for (PayeeACHAccount payee : payees) {
+			LOG.debug("deleting payee = " + payee);
+		}
+		
+		businessObjectService.delete(payees);
+	}
+
+	protected String validateTypeUsedInFile(String inputFileName) {
+		BufferedReader bufferedFileReader = null;
+		String firstType = KFSConstants.EMPTY_STRING;
+		
+		try {
+			try {
+				bufferedFileReader = new BufferedReader(new InputStreamReader(new FileInputStream(inputFileName), "UTF-8"));
```

Comment:
We might want to drop this down into the next try, so that the finally can manage this reader. If we add a finally right here, of course it will blow up when we try to use it below. As it stands, if an exception occurs that is not one of the two in the catches, then this becomes a leaked resource, _and_ line 180 blows up.